Packs/QRadar/Integrations/QRadar_v2/QRadar_v2.py

```diff
@@ -0,0 +1,1961 @@
+    return offense
+
+
+def perform_offense_events_enrichment(
+    offense, additional_where, events_columns, events_limit, client: QRadarClient,
+):
+    """
+    Performs an offense enrichment by:
+        try to create events search via AQL
+        if search created successfully:
+            try to get search results
+            if successfully:
+                enrich offense with event
+        return offense
+    """
+    if is_reset_triggered(client.lock):
+        return offense
+
+    offense_start_time = offense["start_time"]
+    query_expression = (
+        f'SELECT {events_columns} FROM events WHERE INOFFENSE({offense["id"]})'
+        f"{additional_where} limit {events_limit} START '{offense_start_time}'"
+    )
+    events_query = {"headers": "", "query_expression": query_expression}
+    print_debug_msg(
+        f'Starting events fetch for offense {offense["id"]}.', client.lock
+    )
+    try:
+        query_status, search_id = try_create_search_with_retry(client, events_query, offense)
+        offense["events"] = try_poll_offense_events_with_retry(client, offense["id"], query_status, search_id)
+    except Exception as e:
+        print_debug_msg(
+            f'Failed fetching event for offense {offense["id"]}: {str(e)}.',
+            client.lock,
+        )
+    finally:
+        return offense
+
+
+def try_poll_offense_events_with_retry(
+    client, offense_id, query_status, search_id, max_retries=None
+):
+    """
+    Polls search until the search is done (completed/canceled/error), and then returns the search result
+    will retry up to max_retries consecutive failures
+    """
+    if not max_retries:
+        max_retries = EVENTS_FAILURE_LIMIT
+    i = 1
+    failures = 0
+    while not (query_status in TERMINATING_SEARCH_STATUSES or failures >= max_retries):
+        try:
+            if is_reset_triggered(client.lock):
+                return []
+
+            raw_search = client.get_search(search_id)
+            query_status = raw_search.get("status")
+            # failures are relevant only when consecutive
+            failures = 0
+            if query_status in TERMINATING_SEARCH_STATUSES:
+                raw_search_results = client.get_search_results(search_id)
+                print_debug_msg(
+                    f"Events fetched for offense {offense_id}.", client.lock
+                )
+                events = raw_search_results.get("events", [])
+                for event in events:
+                    try:
+                        for time_field in EVENT_TIME_FIELDS:
+                            if time_field in event:
+                                event[time_field] = epoch_to_iso(event[time_field])
+                    except TypeError:
+                        continue
+                return events
+            else:
+                # prepare next run
+                i = i % 60 + EVENTS_INTERVAL_SECS
+                if i >= 60:  # print status debug every fetch sleep (or after)
```

Comment:
not sure how the addition of the events interval make sense here. lets talk about it.

Reply:
This code is intended to print a message every ~minute. It adds `EVENTS_INTERVAL_SECS` to increase the counter by the amount of time the fetch slept, and then checks if we crossed 60 seconds. On the next iteration, it executes `% 60` to move back the counter. 

I'll replace this implementation to a `time.now()` based check, so it'll be more readable and better ensure a message print every minute.